### DISCUSSION TITLE
Invalid client id for oauth login

### DISCUSSION THREAD

**Discussion by MaddyMastering** [dexidp/dex#2460] [Q&A]
Hi, i am trying to implement oauth2 (authorization code flow) using dex.

i can login using static email/password configured, but not able login with github or microsoft

**docker image:** ghcr.io/dexidp/dex:v2.31.0

**docker run command" docker run --net=host -v ./config.docker.yaml:/etc/dex/config.docker.yaml ghcr.io/dexidp/dex:v2.31.0

**config.docker.yaml**
issuer: http://127.0.0.1:5556/dex

storage:
  type: memory

web:
  http: 0.0.0.0:5556

staticClients:
  - id: example-app
    redirectURIs:
            - http://localhost:3000/callback
    name: 'Example App'
    secret: ZXhhbXBsZS1hcHAtc2VjcmV0

oauth2:
  responseTypes: ["code", "token", "id_token"]
  skipApprovalScreen: true

connectors:
  - type: github
    id: github
    name: Github
    config:
      clientID: "***"
      clientSecret: "***"
      redirectURI: "http://localhost:3000/callback"
      orgs:
      - name: super-org
        teams:
        - team-red
  - type: microsoft
    id: microsoft
    name: Microsoft
    config:
      clientID: "***"
      clientSecret: "***"
      tenant: "organizations"
      redirectURI: "http://localhost:3000/callback"

enablePasswordDB: true

staticPasswords:
  - email: "admin@example.com"
    hash: "$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W"
    username: "admin"
    userID: "08a8684b-db88-4b73-90a9-3cd1661f5466"

am i missing anything here ?

**Comment by sagikazarmark** [comment]
@MaddyMastering you may have accidentally leaked credentials. I removed them for now, but you should rotate those creds.
